packages/fiori/src/Wizard.ts

```diff
@@ -995,21 +1014,21 @@ class Wizard extends UI5Element {
 	/**
 	 * Switches the selection from the old step to the newly selected step.
 	 *
-	 * @param {HTMLElement} selectedStep the old step
-	 * @param {HTMLElement} stepToSelect the step to be selected
+	 * @param {WizardStep} selectedStep the old step
+	 * @param {WizardStep} stepToSelect the step to be selected
 	 * @param {Integer} stepToSelectIndex the index of the newly selected step
 	 * @param {boolean} changeWithClick the selection changed due to user click in the step navigation
 	 * @private
 	 */
-	switchSelectionFromOldToNewStep(selectedStep, stepToSelect, stepToSelectIndex, changeWithClick) {
+	switchSelectionFromOldToNewStep(selectedStep: WizardStep | null, stepToSelect: WizardStep, stepToSelectIndex: number, changeWithClick: boolean) {
 		if (selectedStep && stepToSelect) {
 			// keep the selection if next step is disabled
 			if (!stepToSelect.disabled) {
 				selectedStep.selected = false;
 				stepToSelect.selected = true;
 			}
 
-			this.fireEvent("step-change", {
+			this.fireEvent<WizardChangeEventDetails>("step-change", {
```

Comment:
Also export the event detail type for outside usage, someone binding for our events can make use of it:
```js
export type {
   WizardStepChangeEventDetail,
};
```